packages/lib-user/src/components/shared/AllProjects/Projects/Projects.js

```diff
@@ -0,0 +1,118 @@
+import { Box, Text } from 'grommet'
+import { useRef } from 'react'
+import styled from 'styled-components'
+import { arrayOf, bool, func, number, shape, string } from 'prop-types'
+import ProjectCard from '@zooniverse/react-components/ProjectCard'
+import Loader from '@zooniverse/react-components/Loader'
+
+import { useTranslation } from '../../../../translations/i18n.js'
+import { Pagination } from '@components/shared'
+
+const StyledBox = styled(Box)`
+  list-style: none;
+  margin-block-start: 0;
+  padding-inline-start: 0;
+  column-gap: 10px;
+  row-gap: 10px;
+`
+
+const DEFAULT_HANDLER = () => {}
+
+function Projects({
+  error,
+  loading = false,
+  numProjects = 0,
+  page = 1,
+  pageSize = 20,
+  renderedProjects = [],
+  setPage = DEFAULT_HANDLER
+}) {
+  const { t } = useTranslation()
+  const firstCardRef = useRef(null)
+
+  function handlePageChange({ page }) {
+    setPage(page)
+  }
+
+  return (
+    <>
+      {loading ? (
+        <Box fill align='center' justify='center'>
+          <Loader />
+        </Box>
+      ) : error ? (
+        <Box fill align='center' pad='medium'>
+          <Text>{t('AllProjects.error')}</Text>
+        </Box>
+      ) : numProjects === 0 ? (
+        <Box fill align='center' pad='medium'>
+          <Text>{t('AllProjects.noProjects')}</Text>
+        </Box>
+      ) : (
+        <Box align='center'>
+          <StyledBox forwardedAs='ul' direction='row' wrap justify='center'>
+            {renderedProjects?.length && (
+              <li key={renderedProjects?.[0].id} ref={firstCardRef}>
+                <ProjectCard
+                  badge={renderedProjects?.[0].count}
+                  description={renderedProjects?.[0].description}
+                  displayName={renderedProjects?.[0].display_name}
+                  href={`https://www.zooniverse.org/projects/${renderedProjects?.[0].slug}`}
+                  imageSrc={renderedProjects?.[0].avatar_src}
+                  size='small'
+                />
+              </li>
+            )}
+            {renderedProjects?.length && renderedProjects?.slice(1).map(project => {
+              return (
+                <li key={project?.id}>
+                  <ProjectCard
+                    badge={project?.count}
+                    description={project?.description}
+                    displayName={project?.display_name}
+                    href={`https://www.zooniverse.org/projects/${project?.slug}`}
+                    imageSrc={project?.avatar_src}
+                    size='small'
+                  />
+                </li>
+              )
+            })}
```

Comment:
This could be for subsequent PR that adds focus management, but I think this could be simplified  along the lines of:
```
{renderedProjects?.map((project, index) => (
  <li
    key={project?.id}
    ref={index === 0 ? firstCardRef : null}
  >
    <ProjectCard
      badge={project?.count}
      description={project?.description}
      displayName={project?.display_name}
      href={`https://www.zooniverse.org/projects/${project?.slug}`}
      imageSrc={project?.avatar_src}
      size='small'
    />
  </li>
))}
```

Reply:
True. Also agreed about hooking into changes to `renderedProjects` in order to focus on the first card. I think in the case of a `useEffect` I'd set the dependency explicitly to `firstCardRef.current` since that's the one that matters. I'll add focus management in my subsequent PR for sorting projects via `/project_preferences`.

Reply:
By the way, refs aren’t reactive values in React, so can’t be used in the dependency arrays for hooks. ESLint ought to catch that, if you lint your changes.